src/User/UserAuthenticator.php

```diff
@@ -142,8 +152,8 @@ public function setSessionKeysToNullify(array $session_keys) {
    */
   public function nullifySessionKeys() {
     if (!empty($this->sessionKeys)) {
-      array_walk($this->sessionKeys, function ($session_key) {
-        $this->dataHandler->set($this->dataHandler->getSessionPrefix() . $session_key, NULL);
+      array_walk($this->sessionKeys, function ($value, $session_key) {
```

Comment:
We don't pass a key => value pair to this function. `$sessionKeys` is supposed to be a simple array such as `['key1', 'key2']`. Thus, this change has to be undone.